R/find_and_remove_duplicates.R

```diff
@@ -0,0 +1,143 @@
+#' Remove duplicates
+#'
+#' @description
+#' When removing duplicates, users can specify a set columns to consider with
+#' the 'target_columns' argument.
+#'
+#' @param data A input data frame or linelist.
+#' @param target_columns A vector of column names to use when looking for
+#'    duplicates. When the input data is a `linelist` object, this
+#'    parameter can be set to `linelist_tags` if you wish to look for duplicates
+#'    on tagged columns only. Default is `NULL`.
+#' @param remove A vector of duplicate indices to be removed. Duplicate indices
+#'    are unique identifiers for all rows in the original data frame or linelist
+#'    that are duplicates of each other based on the `target_columns`.
+#'    If remove = `NULL` (default value), the first duplicate is kept and
+#'    the rest of the duplicates in the group are removed.
```

Comment:
I don't really understand this argument. The case where the user specifies a numeric value seems like a different functionality, unrelated to duplicate detection, since it would require them to already know where the duplicates are.

Reply:
Good question ...

This argument assumes 2 things:

* the user is looking for duplicates using the `find_duplicates()` function,
* the user is looking for duplicates across few columns (say across tagged columns only)

The result from `find_duplicates()` here will tell rows in the data where individuals are duplicated across the specified columns. However, the information in other columns can be different. If the user decide to maintain the second or third or Nth duplicate due to some information in other columns, this argument will be used in `remove_duplicates()` to tell which one of the duplicates to retain or drop.

Reply:
@Karim-Mane would you be able to provide an example of how the `remove` argument would look like for second/third/second to last index of the duplicate?

This is partly confusing because how to use this argument is unclear.